app/app.go

```diff
@@ -498,6 +502,17 @@ func (app *AkashApp) registerUpgradeHandlers() {
 		return app.mm.RunMigrations(ctx, app.configurator, fromVM)
 	})
 
+	// TODO (merge): fix upgrade handler
+	app.keeper.upgrade.SetUpgradeHandler("akash_v0.15.0", func(ctx sdk.Context,
+		plan upgradetypes.Plan, fromVM module.VersionMap) (module.VersionMap, error) {
+		migrations.MigrateFractionalUakt(ctx,
+			app.keeper.deployment.(dkeeper.Keeper),
+			app.keeper.escrow.(ekeeper.Keeper),
+			app.keeper.market.(mkeeper.Keeper))
+
+		return app.mm.RunMigrations(ctx, app.configurator, fromVM)
+	})
+
```

Comment:
I still need to move the upgrade logic for the corresponding modules to the right places. Instead of it lying in the `app/migrations` package, it should go to `legacy` package of each module.

Reply:
This is done.